### PR TITLE
Matches initial tests and conversion to solkit

### PR DESCRIPTION
**Pull request by jackrieck**
# Summary
- Converted Matches lib to sol-kit
- Added program tests for matches
- Added logic to CI build to skip CI builds when PR is in draft
- Small change to not run CI if PR is in Draft to save build minutes

### CODE REVIEW

File: js/lib/src/state/matches.ts [hunk 1]
```diff
@@ -33,6 +33,23 @@ export enum MatchState {
   Deactivated,
 }
 
+export function convertMatchState(ms: MatchState): {} {
```
Comment (long-banana) [inline js/lib/src/state/matches.ts:36]: I don't see where you are actually calling this function. I do see you importing it into the contract, but I don't see it being used anywhere.

**Review by long-banana — CHANGES_REQUESTED** [review]
Shit. Looks like matches is still pre-(sol-kit). Will want to update that at some point.

What is this empty file doing? [rust/tests/utils.ts](https://github.com/raindrops-protocol/raindrops/pull/50/files#diff-4eea2f85e9c464a4936b0ae5d4d3b3c92cfa549fd942cfc5fb67cffd0ca350c6)

**Comment by jackrieck** [comment]
Yeah having a bit of trouble over all getting this matches lib to cooperate, im marking this as a draft as it's nowhere near ready

File: .github/workflows/raindrops-ci.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ defaults:
 
 jobs:
   build:
+    if: ${{ !github.event.pull_request.draft }}
```
Comment (long-banana) [inline .github/workflows/raindrops-ci.yml:15]: Awesome!

File: js/cli/src/matches.ts [hunk 3]
```diff
@@ -581,3 +572,22 @@ function setLogLevel(value, prev) {
 }
 
 program.parse(process.argv);
+
+async function initMatchesProgram(
```
Comment (long-banana) [inline js/cli/src/matches.ts:573]: There is already a function for this in sol-command. https://github.com/raindrops-protocol/raindrops/blob/81c0a8ef80a1cd7bd11393e3555d8da2253b8fb0/js/cli/src/item.ts#L26

Comment (long-banana) [reply js/cli/src/matches.ts:573]: Below that line is also the preferred way to create the program object from the CLI.

File: js/cli/src/matches.ts [hunk 4]
```diff
@@ -69,7 +69,7 @@ programCommand("create_match")
       );
     }
 
-    await anchorProgram.createMatch(createMatchArgs, {}, config.oracleState);
```
Comment (long-banana) [inline js/cli/src/matches.ts:72]: No more oracleState?

File: js/lib/src/instructions/matches.ts [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+  }
+  async createMatch(
+    args: CreateMatchArgs,
+    _accounts = {},
+    _additionalArgs = {}
+  ) {
+    const [match, _matchBump] = await getMatch(args.winOracle);
+
+    transformTokenValidations(args);
+    return {
```
Comment (long-banana) [inline js/lib/src/instructions/matches.ts:149]: I like this return type format. We should formalize it into sol-kit and update the rest of raindrops at some point.

File: js/lib/src/utils/connection.ts [hunk 6]
```diff
@@ -17,6 +17,10 @@ export const CLUSTERS: Cluster[] = [
     name: "devnet",
     url: clusterApiUrl("devnet"),
   },
+  {
```
Comment (long-banana) [inline js/lib/src/utils/connection.ts:20]: I think matches might have been the last contract that was using this util. This function exists in sol-kit already with localnet.

File: js/lib/src/state/matches.ts (hunk 1 above)
Comment (jackrieck) [reply js/lib/src/state/matches.ts]: Deleted this

**Review by long-banana — CHANGES_REQUESTED** [review]

File: .github/workflows/raindrops-ci.yml (hunk 2 above)
Comment (jackrieck) [reply .github/workflows/raindrops-ci.yml:15]: Another thing I'd like to do is make more of the build steps parallel, like lint/format/build can all just run at the same time IMO

Comment (jackrieck) [reply .github/workflows/raindrops-ci.yml:15]: I get antsy waiting around lol

File: js/cli/src/matches.ts (hunk 4 above)
Comment (jackrieck) [reply js/cli/src/matches.ts:72]: Its used in the args, but the last 2 args to createMatch weren't actually being used so I abstracted them away.

https://github.com/raindrops-protocol/raindrops/blob/81c0a8ef80a1cd7bd11393e3555d8da2253b8fb0/js/cli/src/matches.ts#L38-L49

**Review by long-banana — APPROVED** [review]

**Pull request merged**